apps/viewer/src/app.tsx

```diff
@@ -1,29 +1,34 @@
-import { Box, Flex, HStack, Heading, Text, VStack } from '@chakra-ui/react';
+import { Box, Flex, Heading, HStack, Text, VStack } from '@chakra-ui/react';
 import { StreamStats } from '@millicast/sdk';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
 import ActionBar from '@millicast-react/action-bar';
-import ControlBar from '@millicast-react/control-bar';
-import { IconCameraOff, IconCameraOn, IconSpeaker, IconSpeakerOff } from '@millicast-react/dolbyio-icons';
 import InfoLabel from '@millicast-react/info-label';
 import ParticipantCount from '@millicast-react/participant-count';
 import Timer from '@millicast-react/timer';
 import useNotification from '@millicast-react/use-notification';
-import useViewer from '@millicast-react/use-viewer';
+import useViewer, { SimulcastQuality, SourceId } from '@millicast-react/use-viewer';
 
 import ViewerVideoView from './components/viewer-video-view';
 
 import './styles/font.css';
 
-type TrackSourceState = {
-  muteAudio: boolean;
-  hideVideo: boolean;
-};
-
-type TrackSourcesStates = Map<string, TrackSourceState>;
+const MAX_SOURCES = 4;
 
 function App() {
-  const [trackSourcesStates, setTrackSourcesStates] = useState<TrackSourcesStates>(new Map());
+  const href = new URL(window.location.href);
+  const streamName = href.searchParams.get('streamName') ?? import.meta.env.VITE_MILLICAST_STREAM_NAME;
+  const streamAccountId = href.searchParams.get('streamAccountId') ?? import.meta.env.VITE_MILLICAST_STREAM_ID;
+
+  const { showError } = useNotification();
+
+  const { projectToMainStream, remoteTrackSources, setSourceQuality, startViewer, stopViewer, viewerCount } = useViewer(
+    { streamName, streamAccountId, handleError: showError }
+  );
+
+  const hasInitialisedMainStreamRef = useRef<boolean>(false);
```

Comment:
We don't need this if we could have the `mainStream` instance from useViewer hook, when `mainStream` is undefined, it means not initialised